test_runner/regress/test_compaction.py

```diff
@@ -63,7 +63,9 @@ def test_pageserver_compaction_smoke(neon_env_builder: NeonEnvBuilder):
             log.info(f"Running churn round {i}/{churn_rounds} ...")
 
         workload.churn_rows(row_count, env.pageserver.id)
-        ps_http.timeline_compact(tenant_id, timeline_id)
+        # Force L0 compaction to ensure the number of layers is within bounds; we don't want to count L0 layers
+        # in this benchmark. In other words, this smoke test ensures number of L1 layers are bound.
+        ps_http.timeline_compact(tenant_id, timeline_id, force_l0_compaction=True)
```

Comment:
Can we add an asertion that `l0_layer_count == 0`?

Reply:
There might be background flushes and create new l0 layers, so I don't feel it would make sense to assert it; so I added an assertion that l0 layers <= 1